src/launchpad/kafka.py

```diff
@@ -29,234 +29,92 @@
+        def process_message(msg: Message[KafkaPayload]) -> Any:
             try:
-                # Create Arroyo consumer - minimal config for Arroyo
-                # TODO: When we're closer to production, we'll need a way to disable this logic as
-                # topics, partitions and kafka clusters are configured through getsentry/ops.
-                # We will work with the streaming teams to get this set up.
-                consumer_config = {
-                    "bootstrap.servers": self.bootstrap_servers,
-                    "group.id": self.group_id,
-                    "auto.offset.reset": "latest",
-                    "enable.auto.commit": False,  # Arroyo manages commits via strategies
-                    "enable.auto.offset.store": False,  # Arroyo manages offset storage
-                }
-
-                # Create topics list for Arroyo
-                arroyo_topics = [Topic(topic) for topic in self.topics]
-
-                arroyo_consumer = ArroyoKafkaConsumer(consumer_config)
-
-                # Create processing strategy factory
-                strategy_factory = MessageProcessingStrategyFactory(self.message_handler)
-
-                # Create stream processor
-                self._processor = StreamProcessor(
-                    consumer=arroyo_consumer,
-                    topic=arroyo_topics[0] if arroyo_topics else Topic("default"),
-                    processor_factory=strategy_factory,
-                )
-
-                logger.info("Arroyo Kafka consumer started, calling processor.run()...")
-
-                # This blocks until shutdown is signaled (exactly like Snuba)
-                self._processor.run()
-
-                logger.info("Processor.run() completed")
-                break
-
+                decoded = PREPROD_ARTIFACT_SCHEMA.decode(msg.payload.value)
+                return self.message_handler(decoded)  # type: ignore[no-any-return]
             except Exception as e:
-                if not self._shutdown_requested:
-                    logger.error(f"Error in Kafka consumer: {e}", exc_info=True)
-                    # Sleep a bit before retrying (like Snuba)
-                    time.sleep(1.0)
-                else:
-                    # This else branch is theoretically unreachable due to loop condition
-                    logger.info(  # type: ignore[unreachable]
-                        "Kafka consumer shutting down due to error during shutdown"
-                    )
-                    break
-
-        logger.info("Kafka consumer run loop completed")
+                logger.error(f"Failed to decode message: {e}")
+                raise  # Re-raise the exception to prevent processing invalid messages
+
+        return RunTaskInThreads(
+            processing_function=process_message,
+            concurrency=self.concurrency,
+            max_pending_futures=self.max_pending_futures,
+            next_step=next_step,
+        )
+
+
+def create_kafka_consumer(
```

Comment:
nit: can `create_kafka_consumer` be first in the file? seems like `LaunchpadStrategyFactory` isn't used externally anywhere.